### PR TITLE
fix: Add logic check to allow visible water under other blocks

### PR DESCRIPTION
**Pull request by cvennel**
### Contains

Fixes bug #3889

### How to test

Verify water surface is visible when under other block as in below photos:

Before:
Before

After
after

### CODE REVIEW

File: engine/src/main/java/org/terasology/rendering/primitives/BlockMeshGeneratorSingleShape.java [hunk 1]
```diff
@@ -142,6 +142,10 @@ private boolean isSideVisibleForBlockTypes(Block blockToCheck, Block currentBloc
             return false;
         }
 
+        if (currentBlock.isWater() && (side == Side.TOP) && !blockToCheck.isWater()){
```
Comment (skaldarnar) [inline engine/src/main/java/org/terasology/rendering/primitives/BlockMeshGeneratorSingleShape.java:145]: Please add a `//TODO: ...` comment here to indicate that this only fixes part of a broader problem:

```suggestion
        //TODO: This only fixes the "water under block" issue of the top side not being rendered.
        //      Find a more general way to handle this for "engine:loweredCube" shapes.
        if (currentBlock.isWater() && (side == Side.TOP) && !blockToCheck.isWater()){
```

Comment (cvennel) [reply engine/src/main/java/org/terasology/rendering/primitives/BlockMeshGeneratorSingleShape.java:148]: Thanks Skaldarnar, good point; I've added the TODO. I also put a note about IntelliJ warning `!blockToCheck.isLiquid()` always returns true

**Comment by pollend** [comment]
@Cervator @skaldarnar if there isn't anything else we should merge this.

**Comment by Cervator** [comment]
LGTM and I wouldn't mind merging, just am already up too late for the day :-)

Would be nice to fix further but that's probably beyond this PR. The TODO may also be enough for the reminder rather than making a full separate issue.

**Comment by pollend** [comment]
The existing issue will still be open so it should be fine #3889

**Pull request merged**